components/reddit/sources/new-hot-post-on-a-subreddit/new-hot-post-on-a-subreddit.js

```diff
@@ -0,0 +1,184 @@
+const reddit = require("../../reddit.app.js");
+
+module.exports = {
+  key: "new-hot-spot-on-a-subreddit",
+  name: "New hot spot on a subreddit",
+  description:
+    "Emits an event each time a new host post is added to the top 10 a subreddit.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "The subreddit you'd like to watch for new hot posts.",
+      default: "redditdev",
+    },
+    g: {
+      type: "string",
+      label: "g",
+      description:
+        "Reddit's culture localization on which you'd like to watch for new hot posts.",
+      options: [
+        "GLOBAL",
+        "US",
+        "AR",
+        "AU",
+        "BG",
+        "CA",
+        "CL",
+        "CO",
+        "HR",
+        "CZ",
+        "FI",
+        "FR",
+        "DE",
+        "GR",
+        "HU",
+        "IS",
+        "IN",
+        "IE",
+        "IT",
+        "JP",
+        "MY",
+        "MX",
+        "NZ",
+        "PH",
+        "PL",
+        "PT",
+        "PR",
+        "RO",
+        "RS",
+        "SG",
+        "ES",
+        "SE",
+        "TW",
+        "TH",
+        "TR",
+        "GB",
+        "US_WA",
+        "US_DE",
+        "US_DC",
+        "US_WI",
+        "US_WV",
+        "US_HI",
+        "US_FL",
+        "US_WY",
+        "US_NH",
+        "US_NJ",
+        "US_NM",
+        "US_TX",
+        "US_LA",
+        "US_NC",
+        "US_ND",
+        "US_NE",
+        "US_TN",
+        "US_NY",
+        "US_PA",
+        "US_CA",
+        "US_NV",
+        "US_VA",
+        "US_CO",
+        "US_AK",
+        "US_AL",
+        "US_AR",
+        "US_VT",
+        "US_IL",
+        "US_GA",
+        "US_IN",
+        "US_IA",
+        "US_OK",
+        "US_AZ",
+        "US_ID",
+        "US_CT",
+        "US_ME",
+        "US_MD",
+        "US_MA",
+        "US_OH",
+        "US_UT",
+        "US_MO",
+        "US_MN",
+        "US_MI",
+        "US_RI",
+        "US_KS",
+        "US_MT",
+        "US_MS",
+        "US_SC",
+        "US_KY",
+        "US_OR",
+        "US_SD",
+      ],
+      default: "GLOBAL",
+    },
+    show: {
+      type: "string",
+      label: "show",
+      description:
+        'If "all" is passed, filters such as "hide links that I have voted on" will be disabled.',
+      async options() {
+        return ["none", "all"];
+      }, //QUESTION: How can I have a "null" option, just added the method to "sense" how it is
```

Comment:
Re: this question, are you interested in the case where this param holds no value? Can you make it optional in that case and check to see if `this.show` has a value?

Or were you asking about something else?

Reply:
Ah okay, made it so. 
Was just curious to know if a null or undefined value can be made available in the options for the prop.